tests/test_application.py

```diff
@@ -267,3 +268,41 @@ def helper(scopes, account, authority, *args, **kwargs):
         with self.assertRaises(ExpectedBehavior):
             self.app.acquire_token_silent(["different scope"], self.account)
 
+
+class TestApplicationForClientCapabilities(unittest.TestCase):
+
+    def test_capabilities_and_id_token_claims_merge(self):
+        client_capabilities = ["llt", "ssm"]
+        claims = '''{"id_token": {"auth_time": {"essential": true}}}'''
+        merged_claims = '''{"id_token": {"auth_time": {"essential": true}}, 
+                        "access_token": {"xms_cc": {"values": ["llt", "ssm"]}}}'''
+        # Comparing  dictionaries as JSON object order differs based on python version
+        self.assertEqual(
+            json.loads(merged_claims),
+            json.loads(_merge_claims_and_capabilities(client_capabilities, claims)))
+
+    def test_capabilities_and_id_token_claims_and_access_token_claims_merge(self):
+        client_capabilities = ["llt", "ssm"]
+        claims = '''{"id_token": {"auth_time": {"essential": true}}, 
+                 "access_token": {"nbf":{"essential":true, "value":"1563308371"}}}'''
+        merged_claims = '''{"id_token": {"auth_time": {"essential": true}},
+                        "access_token": {"nbf": {"essential": true, "value": "1563308371"},
+                                        "xms_cc": {"values": ["llt", "ssm"]}}}'''
+        # Comparing  dictionaries as JSON object order differs based on python version
+        self.assertEqual(
+            json.loads(merged_claims),
+            json.loads(_merge_claims_and_capabilities(client_capabilities, claims)))
+
+    def test_no_capabilities_only_claims_merge(self):
+        claims = '''{"id_token": {"auth_time": {"essential": true}}}'''
+        self.assertEqual(claims, _merge_claims_and_capabilities(None, claims))
```

Comment:
The existing assertion is logically wrong. Let's be consistent to use your `json.loads(...)` pattern (which is great! Thanks!).
```suggestion
        self.assertEqual(
            json.loads(...),
            json.loads(_merge_claims_and_capabilities(...))
```